config/v1/types_infrastructure.go

```diff
@@ -47,6 +47,11 @@ type InfrastructureStatus struct {
 	// not support that platform.
 	Platform PlatformType `json:"platform,omitempty"`
 
+	// platformStatus holds status informations that are specific to the underlying
+	// infrastructure provider of the cluster.
+	// +optional
+	PlatformStatus *PlatformStatus `json:"platformStatus,omitempty"`
```

Comment:
Need pointer here?

Reply:
> Need pointer here?

Pointer is so we don't render `platformStatus: {}` for platforms that don't need a status, golang/go#22480.

Reply:
Continuing from [here][1] since this thread isn't outdated.

> See as an example `oc get image.config.openshift.io/cluster` - structs like `RegistrySources` don't render if they aren't set.

Indeed the YAML output does not include the empty structure:

```console
$ KUBECONFIG=cluster-bot-2019-04-29-204922.kubeconfig oc get -o yaml image.config.openshift.io/cluster
apiVersion: config.openshift.io/v1
kind: Image
metadata:
  annotations:
    release.openshift.io/create-only: "true"
  creationTimestamp: "2019-04-29T21:05:30Z"
  generation: 1
  name: cluster
  resourceVersion: "12590"
  selfLink: /apis/config.openshift.io/v1/images/cluster
  uid: 856cda66-6ac2-11e9-9f29-128efb056b1e
spec: {}
status:
  internalRegistryHostname: image-registry.openshift-image-registry.svc:5000
```

And cranking up the debug logs:

```console
$ KUBECONFIG=cluster-bot-2019-04-29-204922.kubeconfig oc -v10 get -o yaml image.config.openshift.io/cluster
...
I0429 15:24:35.123646   24641 request.go:942] Response Body: {"apiVersion":"config.openshift.io/v1","kind":"Image","metadata":{"annotations":{"release.openshift.io/create-only":"true"},"creationTimestamp":"2019-04-29T21:05:30Z","generation":1,"name":"cluster","resourceVersion":"12590","selfLink":"/apis/config.openshift.io/v1/images/cluster","uid":"856cda66-6ac2-11e9-9f29-128efb056b1e"},"spec":{},"status":{"internalRegistryHostname":"image-registry.openshift-image-registry.svc:5000"}}
...
```

so whatever does the `+optional` trimming must live on the server side.  So the issue here is just whether we care about these optional properties being rendered as `{}` for folks using `json.Marshal` and `ghodss/yaml`; folks who consume from the API itself will be fine.

[1]: https://github.com/openshift/api/pull/300#discussion_r279492177

Reply:
Per upstream [contributing docs](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required), pointer + `omitempty` + `// +optional` is the right way to go here.